### PR TITLE
Update Libraries section

### PR DESCRIPTION
**Pull request by stjepanbanek**
Dear and respected colleague - Mr. Filip Floreani. 
This is what I came up with while updating the libs section. Removed some outdated ones, added some new ones.
Hope you like it. 
One note: I kept the ThreeTen lib on the list even though it's not updated anymore, because Im sure it's still widely used.
As you probably know Date API from Java 8 in now available natively in Andoird. With some tweaks, of course :)

### CODE REVIEW

File: Useful tools & utilities/Libraries.md [hunk 1]
```diff
@@ -1,28 +1,22 @@
...
+3. **MPAndroidChart**
 
-	Glide is a fast and efficient open source media management and image loading framework for Android. It wraps media decoding, memory and disk caching, and resource pooling into a simple and easy-to-use interface. It also has support for animated GIFs.
+    MPAndroidChart is a powerful graph view library supporting line, pie, bar radar, bubble graphs and much more. It also includes other features like scaling, panning and animations.
 
-	More info with code examples is available [here](https://github.com/bumptech/glide).
+    More info with code examples is available [here](https://github.com/PhilJay/MPAndroidChart).
 
 
 ## Networking
```
Comment (ffloreani) [inline Useful tools & utilities/Libraries.md:22]: I recommend we remove GSON from this list, as we're not using it in any new projects, and just stick with Moshi 🙂

File: Useful tools & utilities/Libraries.md [hunk 2]
```diff
@@ -55,29 +49,23 @@
...
+1. **Room**
 
 	Room is a library that provides an abstraction layer over the SQLite database. It supports all SQL functionalities.
 
 	More info with code examples is available [here](https://developer.android.com/topic/libraries/architecture/room.html).
 
 	What's important about this library are migrations. They are crucial if you wish to migrate your database to a new version. If you don't set up migrations, the room will not crash, nor will it log anything. What it will do is **delete all database data and rebuild the database.** You do **not** want this to happen in production, so be careful when using it and set up your migrations properly. If you are not sure how to do it, ask your team lead to help you out.
 
-3. **DbInspector**
+2. **DbInspector**
```
Comment (ffloreani) [inline Useful tools & utilities/Libraries.md:68]: This reminded me - would it be a good idea to open our "In-house libraries" page to the public? Then we could remove them from this page and have them all neatly arranged in a single page for bragging rights. I'd like to hear your opinion on this 🙂

File: Useful tools & utilities/Libraries.md [hunk 3]
```diff
@@ -106,11 +94,33 @@
...
+
+    Hilt is a library built on top of Dagger to simplify the setup process.
+
+    More info on Hilt (recommended) with code examples is available [here](https://developer.android.com/training/dependency-injection/hilt-android).
+
+    For Dagger without Hilt, info and code examples are available [here](https://developer.android.com/training/dependency-injection/dagger-android).
+
+4. **Android KTX**
+
+    Android KTX is a set of Kotlin extensions that are included with Android Jetpack and other Android libraries. KTX extensions provide idiomatic Kotlin to Jetpack, Android platform, and other APIs
```
Comment (ffloreani) [inline Useful tools & utilities/Libraries.md:109]:
```suggestion
    Android KTX is a set of Kotlin extensions that are included with Android Jetpack and other Android libraries. KTX extensions provide idiomatic Kotlin to Jetpack, Android platform and other APIs.
```

File: Useful tools & utilities/Libraries.md [hunk 4]
```diff
@@ -106,11 +94,33 @@
...
+
+    Every project written in Kotlin should use these as they make your code more concise, readable and shorter.
+
+    More info with code examples is available [here](https://developer.android.com/kotlin/ktx).
+
+5. **Lifecycle**
+
+    Lifecycle from AndroidX includes everything we need to make a good MVVM application such as: ViewModels, LiveData, Lifecycle callbacks etc.
```
Comment (ffloreani) [inline Useful tools & utilities/Libraries.md:117]:
```suggestion
    Lifecycle from AndroidX includes everything you need to make a good MVVM application, such as: ViewModels, LiveData, Lifecycle callbacks, etc.
```

File: Useful tools & utilities/Libraries.md [hunk 5]
```diff
@@ -106,11 +94,33 @@
...
+
+    It is essential for MVVM arcitecture, but it can also help in lost of other use cases.
```
Comment (ffloreani) [inline Useful tools & utilities/Libraries.md:119]: I'm not sure what you meant by "help in lost of other use cases", could you explain? 😄

File: Useful tools & utilities/Libraries.md [hunk 6]
```diff
@@ -106,11 +94,33 @@
...
 
-	A utility library for making Google Play screenshots. It draws over your status bar, showing only a full battery icon and clock. Use it to get nice, clean screenshots without those pesky empty battery icons and notifications.
+    To see all features and variations of androidx.lifecycle go [here](https://developer.android.com/jetpack/androidx/releases/lifecycle).
```
Comment (ffloreani) [inline Useful tools & utilities/Libraries.md:121]:
```suggestion
    To see all features and variations of androidx.lifecycle, go [here](https://developer.android.com/jetpack/androidx/releases/lifecycle).
```

File: Useful tools & utilities/Libraries.md [hunk 7]
```diff
@@ -106,11 +94,33 @@
...
 
-	More info with code examples is available [here](https://github.com/emmaguy/clean-status-bar).
+    For more info on ViewModel with code examples go [here](https://developer.android.com/topic/libraries/architecture/viewmodel).
```
Comment (ffloreani) [inline Useful tools & utilities/Libraries.md:123]:
```suggestion
    For more info on ViewModel with code examples, go [here](https://developer.android.com/topic/libraries/architecture/viewmodel).
```

**Review by ffloreani — COMMENTED** [review]
This is a nice update and a welcome change, but I was hoping that you'd include a couple more libraries that we use on a regular basis, like:

- Lottie
- ExoPlayer?
- Mockito-Kotlin
- OkHttp MockWebServer
- Espresso

The last 3 would be an especially nice addition, as our testing section currently looks like it was hit by Covid and everyone just ran away from it.

**Comment by stjepanbanek** [comment]
Thanks for comments! You are absolutely right, I should have added more libs, but I just couldn't remember what else to include. Your suggestions will be added :). Espresso though? Do we really use it?

**Comment by ffloreani** [comment]
Not as much as we should, but yeah, Espresso is the go-to tool for UI testing 🙂

File: Useful tools & utilities/Libraries.md (hunk 1 above)
Comment (stjepanbanek) [reply Useful tools & utilities/Libraries.md:22]: Sure!

File: Useful tools & utilities/Libraries.md (hunk 2 above)
Comment (stjepanbanek) [reply Useful tools & utilities/Libraries.md:68]: I don't see why not :). I think we would benefit from advertising our in-house libs. It was one of the company's pros for me when I was applying :D.

File: Useful tools & utilities/Libraries.md [hunk 8]
```diff
@@ -91,6 +83,29 @@
 
 	More info with code examples is available [here](http://mockito.org/).
 
+2. **Mockito Kotlin**
+
+    A small library that introduces kotlin helpers for working with Mockito. If you use Kotlin when writing tests,
```
Comment (ffloreani) [inline Useful tools & utilities/Libraries.md:88]:
```suggestion
    A small library that introduces Kotlin helpers for working with Mockito. If you use Kotlin when writing tests,
```

File: Useful tools & utilities/Libraries.md [hunk 9]
```diff
@@ -91,6 +83,29 @@
...
+
+    this will make them shorter and more readable.
+
+    More info with code examples is available [here](https://github.com/nhaarman/mockito-kotlin).
+
+3. **OkHttp MockWebServer**
+
+    It can be very difficult to mock web requests and responses. This is where MockWebServer comes in handy.
+
+    It provides an ease-to-use scriptable web server implementation that can be used for testing your HTTP client.
```
Comment (ffloreani) [inline Useful tools & utilities/Libraries.md:98]:
```suggestion
    It provides an easy-to-use scriptable web server implementation that can be used for testing your HTTP client.
```

**Review by ffloreani — APPROVED** [review]
Mucho gusto 😍

**Pull request merged**